src/parse.rkt

```diff
@@ -72,6 +72,16 @@
     list-syntax
     list-syntax))
 
+(define/contract (elm-record-stx list-syntax list)
+                 (-> any/c 
+                     (flat-named-contract 'even-sized-list (compose even? length)) 
+                     syntax?)
+  (datum->syntax
+    list-syntax
+    #`(elm-record #,@(sequence->list (in-slice 2 list)))
+    list-syntax
+    list-syntax))
+
 ; {abc def ...} -> (elm-record (abc def) ...)
 (define (read-elm-record ch in src ln col pos)
   (define list-syntax
```

Comment:
Btw, I like to use `let` instead of nested `define`s, seems more functional to me and also better readable because of the indentation it produces. But I know that Racket docs are littered with examples showing nested `define`s so I won't say it's not "the Racket way" ;)